src/NHSD.BuyingCatalogue.Ordering.Api/Extensions/ClaimsPrincipalExtensions.cs

```diff
@@ -17,9 +18,15 @@ public static Guid GetUserId(this ClaimsPrincipal user)
             return new Guid(user.FindFirstValue(ClaimTypes.NameIdentifier));
         }
 
-        public static string GetUserName(this ClaimsPrincipal user)
-        {
-            return user.Identity.Name;
+        public static string GetUserName(this ClaimsPrincipal user)
+        {
+            return user.Identity.Name;
+        }
+        public static void SetLastUpdated(this Order order, ClaimsPrincipal user)
```

Comment:
Have a look at Ruth's branch `feature/4621-Put-Supplier`. The code encapsulates the setting of the last updated date as the API shouldn't need to know to set that every time.

Code use example:

```
            var name = User.Identity.Name;
            order.SetLastUpdatedBy(User.GetUserId(), name);
```